pkg/resources/provider.go

```diff
@@ -67,3 +69,20 @@ func (r *ReconcileResourceProvider) ReconcileResultSecret(ctx context.Context, o
 	}
 	return nil
 }
+
+func IsLastResource(ctx context.Context, c client.Client) (bool, error) {
+	listOptions := client.ListOptions{
+		Namespace: "",
+	}
+	var postgresList = &v1alpha1.PostgresList{}
+	if err := c.List(ctx, postgresList, &listOptions); err != nil {
+		msg := "failed to retrieve postgres cr(s)"
+		return false, errorUtil.Wrap(err, msg)
+	}
```

Comment:
You could exit the function early if the `postgresList` is bigger than 1.

Reply:
I just copied this from one of the providers without looking at it but you're right it needs a change - it actually will only work if this is the last redis instance - I'll change it to work more generally